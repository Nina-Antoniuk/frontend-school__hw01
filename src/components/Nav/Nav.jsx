import React from 'react';
import { NavLink } from 'react-router-dom';
import s from './Nav.module.scss';

const Nav = function Nav() {
  return (
    <nav className={s.Nav}>
      <ul className={s.NavList}>
        <li>
          <NavLink
            exact
            to="/"
            className={s.NavLink}
            activeClassName={s.activeLink}
          >
            News
          </NavLink>
        </li>
        <li>
          <NavLink
            to={`/profile/${'dave.xp'}`}
            className={s.NavLink}
            activeClassName={s.activeLink}
          >
            Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
export default Nav;
