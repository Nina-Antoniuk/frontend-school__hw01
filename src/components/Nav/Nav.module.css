@media screen and (min-width: 768px) {
  .Nav {
    position: fixed;
    min-width: 20%;
    height: 100vh;
    padding: 3vh 2vw;
  }
}

.NavList {
  padding: 0;
  list-style-type: none;
}

@media screen and (max-width: 767.98px) {
  .NavList {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-pack: distribute;
    justify-content: space-around;
  }
}

.NavLink {
  display: block;
  padding: 10px 0;
  font-size: 20px;
  text-decoration: none;
  color: var(--dark-color);
  -webkit-transition: color var(--delay) ease;
  transition: color var(--delay) ease;
}

.NavLink:hover {
  color: var(--main-accent-color);
}

.activeLink {
  color: var(--main-accent-color);
}

.activeLink::after {
  content: '';
  display: block;
  width: 100%;
  height: 2px;
  background-color: var(--main-accent-color);
}
/*# sourceMappingURL=Nav.module.css.map */
