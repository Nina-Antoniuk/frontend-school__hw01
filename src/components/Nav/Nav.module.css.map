{
    "version": 3,
    "mappings": "AAGE,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,KAAK;EADrC,AAAA,IAAI,CAAC;IAED,QAAQ,EAAE,KAAK;IACf,SAAS,EAAE,GAAG;IACd,MAAM,EAAE,KAAK;IACb,OAAO,EAAE,OAAO;GAEnB;;;AAED,AAAA,QAAQ,CAAC;EACP,OAAO,EAAE,CAAC;EACV,eAAe,EAAE,IAAI;CAMtB;;AAJC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,QAAQ;EAJxC,AAAA,QAAQ,CAAC;IAKL,OAAO,EAAE,IAAI;IACb,eAAe,EAAE,YAAY;GAEhC;;;AAED,AAAA,QAAQ,CAAC;EACP,OAAO,EAAE,KAAK;EACd,OAAO,EAAE,MAAM;EACf,SAAS,EAAE,IAAI;EACf,eAAe,EAAE,IAAI;EACrB,KAAK,EAAE,iBAAiB;EACxB,UAAU,EAAE,KAAK,CAAC,YAAY,CAAC,IAAI;CAKpC;;AAXD,AAQE,QARM,AAQL,MAAM,CAAC;EACN,KAAK,EAAE,wBAAwB;CAChC;;AAGH,AAAA,WAAW,CAAC;EACV,KAAK,EAAE,wBAAwB;CAShC;;AAVD,AAGE,WAHS,AAGR,OAAO,CAAC;EACP,OAAO,EAAE,EAAE;EACX,OAAO,EAAE,KAAK;EACd,KAAK,EAAE,IAAI;EACX,MAAM,EAAE,GAAG;EACX,gBAAgB,EAAE,wBAAwB;CAC3C",
    "sources": [
        "Nav.module.scss",
        "../../shared/scss/variables.scss"
    ],
    "names": [],
    "file": "Nav.module.css"
}