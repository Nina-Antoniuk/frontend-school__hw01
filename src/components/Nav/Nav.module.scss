@import '../../constants/scss/variables.scss';

.Nav {
  @media screen and (min-width: $breakpoint-min-tablet) {
    position: fixed;
    min-width: 20%;
    height: 100vh;
    padding: 3vh 2vw;
  }
}

.NavList {
  padding: 0;
  list-style-type: none;

  @media screen and (max-width: $breakpoint-max-mobile) {
    display: flex;
    justify-content: space-around;
  }
}

.NavLink {
  display: block;
  padding: 10px 0;
  font-size: 20px;
  text-decoration: none;
  color: var(--dark-color);
  transition: color var(--delay) ease;

  &:hover {
    color: var(--main-accent-color);
  }
}

.activeLink {
  color: var(--main-accent-color);

  &::after {
    content: '';
    display: block;
    width: 100%;
    height: 2px;
    background-color: var(--main-accent-color);
  }
}
