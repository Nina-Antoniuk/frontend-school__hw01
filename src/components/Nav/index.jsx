import React from 'react';
import { NavLink } from 'react-router-dom';
import { ROUTES } from 'consts';
import styles from './Nav.module.scss';

const Nav = function Nav() {
  return (
    <nav className={styles.Nav}>
      <ul className={styles.NavList}>
        <li>
          <NavLink
            exact
            to={ROUTES.HOME}
            className={styles.NavLink}
            activeClassName={styles.activeLink}
          >
            News
          </NavLink>
        </li>
        <li>
          <NavLink
            to={`${ROUTES.USER_PROFILE}/${'dave.xp'}`}
            className={styles.NavLink}
            activeClassName={styles.activeLink}
          >
            Profile
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};
export default Nav;
