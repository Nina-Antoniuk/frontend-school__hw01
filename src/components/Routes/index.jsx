import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { ROUTES } from 'consts';
import LoaderComponent from 'components/LoaderComponent';
import NewsFeed from 'pages/NewsFeed';
import Profile from 'pages/Profile';

const Routs = function Routs({ getFirstVideo, firstVideo }) {
  return (
    <div className="content">
      <React.Suspense fallback={<LoaderComponent />}>
        <Switch>
          <Route exact path={ROUTES.HOME}>
            <NewsFeed getVideo={getFirstVideo} />
          </Route>
          <Route path={`${ROUTES.USER_PROFILE}/:id`}>
            <Profile firstVideo={firstVideo} />
          </Route>
          <Route>
            <Redirect exact to={ROUTES.HOME} />
          </Route>
        </Switch>
      </React.Suspense>
    </div>
  );
};

Routs.propTypes = {
  getFirstVideo: PropTypes.func.isRequired,
  firstVideo: PropTypes.object.isRequired,
};
export default Routs;
