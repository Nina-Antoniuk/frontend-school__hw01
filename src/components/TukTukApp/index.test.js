import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import '@testing-library/jest-dom';
import TikTukApp from './index';
import Nav from '../Nav';
import Routes from '../Routes';

jest.mock('../Nav', () => {
  return () => {
    return <p>Mocked Nav component</p>;
  };
});

jest.mock('../Routes', () => {
  return () => {
    return <p>Mocked Routes component</p>;
  };
});

test('Check Routes component rendering', () => {
  const history = createMemoryHistory();
  const container = render(
    <Router history={history}>
      <TikTukApp />
    </Router>,
  );
  expect(container).toMatchSnapshot();
});
