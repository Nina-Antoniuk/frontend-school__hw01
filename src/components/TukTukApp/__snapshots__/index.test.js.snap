// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check Routes component rendering 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdvvtL hFwisg"
      >
        <div
          class="sc-gsDKAQ ewZRyA"
        >
          <header
            class="sc-dkPtRN kKaBkD"
          >
            <p>
              Mocked Nav component
            </p>
            <div />
            <div>
              
    
              <div
                class="wrapper"
                style="display: flex; justify-content: center;"
              >
                
      
                <svg
                  aria-label="Иконка солнца"
                  class="lightThemeIconColor"
                  role="img"
                  style="width: 26px; height: 26px;"
                >
                  
        
                  <use
                    href="sprite.svg#icon-sun"
                  />
                  
      
                </svg>
                
      
                <div
                  class="control"
                >
                  
        
                  <input
                    aria-label="Переключить между тёмной и светлой темой"
                    class="toggle"
                    id="toggle"
                    name="theme"
                    type="checkbox"
                  />
                  
          
        
                  <label
                    aria-hidden="true"
                    class="track lightThemeTrackColor"
                    for="toggle"
                  >
                    
        
                  </label>
                  
        
        
                  <div
                    aria-hidden="true"
                    class="marker"
                  />
                  
      
                </div>
                 

      
                <svg
                  aria-label="Иконка солнца"
                  class="icon lightThemeIconColor"
                  role="img"
                >
                  
        
                  <use
                    href="sprite.svg#icon-moon"
                  />
                  
      
                </svg>
                
    
              </div>
              
  
            </div>
          </header>
          <main>
            <p>
              Mocked Routes component
            </p>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdvvtL hFwisg"
    >
      <div
        class="sc-gsDKAQ ewZRyA"
      >
        <header
          class="sc-dkPtRN kKaBkD"
        >
          <p>
            Mocked Nav component
          </p>
          <div />
          <div>
            
    
            <div
              class="wrapper"
              style="display: flex; justify-content: center;"
            >
              
      
              <svg
                aria-label="Иконка солнца"
                class="lightThemeIconColor"
                role="img"
                style="width: 26px; height: 26px;"
              >
                
        
                <use
                  href="sprite.svg#icon-sun"
                />
                
      
              </svg>
              
      
              <div
                class="control"
              >
                
        
                <input
                  aria-label="Переключить между тёмной и светлой темой"
                  class="toggle"
                  id="toggle"
                  name="theme"
                  type="checkbox"
                />
                
          
        
                <label
                  aria-hidden="true"
                  class="track lightThemeTrackColor"
                  for="toggle"
                >
                  
        
                </label>
                
        
        
                <div
                  aria-hidden="true"
                  class="marker"
                />
                
      
              </div>
               

      
              <svg
                aria-label="Иконка солнца"
                class="icon lightThemeIconColor"
                role="img"
              >
                
        
                <use
                  href="sprite.svg#icon-moon"
                />
                
      
              </svg>
              
    
            </div>
            
  
          </div>
        </header>
        <main>
          <p>
            Mocked Routes component
          </p>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
